backend/LexBoxApi/LexBoxKernel.cs

```diff
@@ -54,6 +54,7 @@ public static class LexBoxKernel
         services.AddScoped<TusService>();
         services.AddScoped<TurnstileService>();
         services.AddScoped<IHgService, HgService>();
+        services.AddHostedService<HgService>();
```

Comment:
How does AddHostedService interact with AddScoped? Will the StartAsync method be called every time a new HgService instance is initialized, i.e. on every request that ASP.NET Core handles? If so, creating the a-z and 0-9 directories would be happening way more often than needed, and an initContainer would be better. If the IHostedService interface is somewhat independent of AddScoped, and StartAsync/StopAsync are only called once per lifetime of the ASP.NET Core process, then this would be fine.

Reply:
It's only called once. It would be a pretty sneaky gotcha if AddScoped and AddHostedService affected each other 🙂